server/services/mapping_service.py

```diff
@@ -96,6 +96,27 @@ def unlock_task_after_mapping(mapped_task: MappedTaskDTO) -> TaskDTO:
         return task.as_dto()
 
     @staticmethod
+    def stop_mapping_task(stop_task: StopMappingTaskDTO) -> TaskDTO:
```

Comment:
Common code between stop and unlock should be refactored into a single method shared by both, to prevent bugs and ease testing.